CMakeLists.txt

```diff
@@ -112,6 +112,90 @@ endif()
 # Build
 # =====
 
+function(PREPEND var prefix)
+   set(listVar "")
+   foreach(f ${ARGN})
+      list(APPEND listVar "${prefix}${f}")
+   endforeach(f)
+   set(${var} "${listVar}" PARENT_SCOPE)
+endfunction()
+
+function(POSTFIX var postfix)
+   set(listVar "")
+   foreach(f ${ARGN})
+      list(APPEND listVar "${f}${postfix}")
+   endforeach(f)
+   set(${var} "${listVar}" PARENT_SCOPE)
+endfunction()
+
+set(XTENSOR_SINGLE_INCLUDE
```

Comment:
I want to simply dump all headers, but the problem is the json dependency that should not be enforced on the user I think

Reply:
Can we reuse the original header list and split those that should not be included by default?

There may be more than just JSON.

Reply:
Thant would be better indeed. Do you have such a list? Another option would be to see if e.g. JSON is defined, but I think that that is less transparent.

Reply:
What I mean is that there is duplication between that list and XTENSOR_HEADERS.